modules/data.remote/R/call_remote_process.R

```diff
@@ -0,0 +1,400 @@
+      PEcAn.DB::db.query(sprintf("UPDATE dbfiles SET file_path='%s', file_name='%s' WHERE container_id=%f;", output$raw_data_path, output$raw_data_name, raw_id), dbcon)      
+    }else if(flag == 4){
+      # requested processed and raw files are present and have to be updated
+      pro_id = pro_check$id
+      raw_id = raw_check$id
+      PEcAn.logger::logger.info("updating processed and raw files")
+      PEcAn.DB::db.query(sprintf("UPDATE inputs SET start_date='%s', end_date='%s', name='%s' WHERE id=%f;", write_pro_start, write_pro_end, output$process_data_name, pro_id), dbcon)
+      PEcAn.DB::db.query(sprintf("UPDATE dbfiles SET file_path='%s', file_name='%s' WHERE container_id=%f;", output$process_data_path, output$process_data_name, pro_id), dbcon)
+      PEcAn.DB::db.query(sprintf("UPDATE inputs SET start_date='%s', end_date='%s', name='%s' WHERE id=%f", write_raw_start, write_raw_end, output$raw_data_name, raw_id), dbcon)
+      PEcAn.DB::db.query(sprintf("UPDATE dbfiles SET file_path='%s', file_name='%s' WHERE container_id=%f;", output$raw_data_path, output$raw_data_name, raw_id), dbcon)
+    }else if(flag == 5){
+      # raw file required for creating the processed file exists and the processed file needs to be updated
+      pro_id = pro_check$id
+      PEcAn.logger::logger.info("Updating the existing processed file")
+      PEcAn.DB::db.query(sprintf("UPDATE inputs SET start_date='%s', end_date='%s', name='%s' WHERE id=%f;", write_pro_start, write_pro_end, output$process_data_name, pro_id), dbcon)
+      PEcAn.DB::db.query(sprintf("UPDATE dbfiles SET file_path='%s', file_name='%s' WHERE container_id=%f;", output$process_data_path, output$process_data_name, pro_id), dbcon)
+    }
+    }
+  }
+  else{
+    # if the requested raw file already exists within the required timeline dont insert or update the DB
+    if(as.character(write_raw_start) == "dont write" && as.character(write_raw_end) == "dont write"){
+      PEcAn.logger::logger.info("Requested raw file already exists")
+    }else{
+    if(flag == 1){
+      PEcAn.logger::logger.info(("Inserting raw file for the first time"))
+      PEcAn.DB::dbfile.input.insert(in.path = output$raw_data_path, in.prefix = output$raw_data_name, siteid = siteid, startdate = write_raw_start, enddate = write_raw_end, mimetype = raw_mimetype, formatname = raw_formatname, con = dbcon)
+    }else{
+      PEcAn.logger::logger.info("Updating raw file")
+      raw_id = raw_check$id
+      PEcAn.DB::db.query(sprintf("UPDATE inputs SET start_date='%s', end_date='%s', name='%s' WHERE id=%f;", write_raw_start, write_raw_end, output$raw_data_name, raw_id), dbcon)
+      PEcAn.DB::db.query(sprintf("UPDATE dbfiles SET file_path='%s', file_name='%s' WHERE container_id=%f;", output$raw_data_path, output$raw_data_name, raw_id), dbcon)
+    }
+    }
+  }
+  
+  PEcAn.DB::db.close(con=dbcon)
+  
```

Comment:
Function exits without returning updated `input` information to the settings file (i.e. the input ids and file paths to the data that's been downloaded). Take a look at the end of `met.process` for an analogy

Also, function appears to strictly assume that the processing is running locally, and doesn't consider the case where the data needs to be downloaded and processed directly onto a remote machine (e.g. HPC).  Similarly, it's not clear how this function behaves if the user is trying to download additional data to append onto the end of an existing time-series. Take a look at `convert.inputs` and `remote.execute.R`.

Reply:
> Similarly, it's not clear how this function behaves if the user is trying to download additional data to append onto the end of an existing time-series.

I have now added some documentation about this which I still have to extend.

> Whenever a data product is requested the output files are stored in the inputs table of BETYdb. Subsequently when the same product is requested again with a different date range but with the same qc, scale, projection the previous file in the db would be extended. The DB would always contain only one file of the same type.
As an example, if a file `Reykajvik_gee_s2_10.0_NA_1.0_200802174519.nc` containing Sentinel 2 bands for start date: 2018-01-01, end date: 2018-06-30 exists in the DB and the same product is requested again for a different date range one of the following cases would happen,

>1. New dates are ahead of the existing file: For example, if the requested dates are start: 2018-10-01, end: 2018-12-31 in this case the previous file will be extended forward meaning the effective start date of the file to be downloaded would be the day after the end date of the previous file record, i.e. 2018-07-01. The new and the previous file would be merged and the DB would now be having data for 2018-01-01 to 2018-12-31.

>2. New dates are preceding of the existing file: For example, if the requested dates are start: 2017-01-01, end: 2017-06-30 in this case the effective end date of the new download would be the day before the start date of the existing file, i.e., 2017-12-31. The new and the previous file would be merged and the file in the DB would now be having data for 2017-01-01 to 2018-06-30. 

>3. New dates contain the date range of the existing file: For example, if the requested dates are start: 2016-01-01, end: 2019-06-30 here the existing file would be replaced entirely with the new file. A more efficient way of doing this could be to divide your request into two parts, i.e, first request for 2016-01-01 to 2018-01-01 and then for 2018-06-30 to 2019-06-30.

Reply:
Great to hear the function is smart about downloads -- what you describe is exactly the behavior we've implemented elsewhere. 👍